bindings/java/src/main/java/com/apple/foundationdb/tuple/TupleUtil.java

```diff
@@ -614,18 +667,35 @@ static boolean hasIncompleteVersionstamp(Stream<?> items) {
 		});
 	}
 
+	static void replace(ByteArrayOutputStream encoded, byte array[], byte search,
```

Comment:
Now that the other `replace` has been updated, is there any reason that this function can't be removed?

Reply:
The other function expects (byte array[], byte search[], byte replace[]), but this one expects (ByteArrayOutputStream encoded, byte array[], byte search, byte replace[])
And when we have ByteArrayOutputStream this function will work faster than the other one, we will just add all bytes into ByteArrayOutputStream instead of creating new byte array and adding it.

Reply:
Hm, but now we have duplicate code of a (fairly intricate) implementation of `replace`, so I don't think we want to keep both of them. Should the `replace` that returns a byte array call the one that takes a `ByteArrayOutputStream` with an output stream it allocates so that we only need to maintain one copy of this function?

Reply:
Oh, I see. But there is a little bit difference, the replace with output stream takes byte for search parameter, the other one takes byte array. I will change it (the search parameter will be byte array), and will call this one from the other replace function.